zookeeper-api/src/main/java/org/apache/helix/zookeeper/zkclient/ZkClient.java

```diff
@@ -1858,24 +1923,47 @@ public Object call() throws Exception {
     });
   }
 
+  private boolean watchForData(final String path, boolean skipWatchingNodeNotExist) {
```

Comment:
code style-wise, we should implement the original watchForData method using this new one to avoid duplicate logic.

Reply:
original watchForData use new one now,